### PR TITLE
Fix render-delay-extension test, which was asserting promise rejection incorrectly.

### PR DESCRIPTION
**Pull request by lannka**

### CODE REVIEW

File: test/functional/test-render-delaying-extensions.js [hunk 1]
```diff
@@ -43,13 +45,15 @@ describe('waitForExtensions', () => {
...
+    window.document.head.innerHTML = '';
   });
 
   it('should return undefined if no extension is presented', () => {
     expect(waitForExtensions(win)).to.equal(undefined);
   });
 
-  it('should keep waiting when some extensions are not ready yet', () => {
+  it('should timeout if some extensions do not load', () => {
+    win = window; // Use the main window to be able to tick the clock
```
Comment (jridgewell) [inline test/functional/test-render-delaying-extensions.js]: You can use `lolex` to install the fake timers into the iframe.

Comment (lannka) [inline test/functional/test-render-delaying-extensions.js]: Haha, that's something I was looking for. Should have asked you folks earlier :-)

**Comment by lannka** [comment]
Done. PTAL

File: test/functional/test-render-delaying-extensions.js [hunk 2]
```diff
@@ -40,20 +40,20 @@ describe('waitForExtensions', () => {
...
     return createIframePromise().then(iframe => {
       win = iframe.win;
+      clock = lolex.install(iframe.win);
     });
   });
 
   afterEach(() => {
     sandbox.restore();
-    window.document.head.innerHTML = '';
+    clock.uninstall();
```
Comment (jridgewell) [inline test/functional/test-render-delaying-extensions.js]: No need, we're killing the iframe.

**Pull request merged**